roles/download/defaults/main.yml

```diff
@@ -60,7 +60,7 @@ downloads:
     url: "{{ calico_cni_download_url }}"
     owner: "root"
     mode: "0755"
-    enabled: "{{ kube_network_plugin == 'calico' or kube_network_plugin == 'canal' }}"
+    enabled: "{{ kube_network_plugin == 'calico' and not use_calico_cni_container|default(false) }}"
```

Comment:
Let's actually remove calico_cni_plugin download resource and only use calico_cni